sound/soc/sof/loader.c

```diff
@@ -192,6 +192,42 @@ static int ext_man_get_dbg_abi_info(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int ext_man_get_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_config_data *config =
+		container_of(hdr, struct sof_ext_man_config_data, hdr);
+	int elems_counter;
+	int elems_size;
+	int i;
+
+	/* calculate elements counter */
+	elems_size = config->hdr.size - sizeof(struct sof_ext_man_elem_header);
+	elems_counter = elems_size / sizeof(struct sof_config_elem);
+
+	dev_dbg(sdev->dev, "ext_man_get_config_data can hold up to %d config elements\n",
+		elems_counter);
+
+	for (i = 0; i < elems_counter; ++i) {
+		dev_dbg(sdev->dev, "ext_man_get_config_data get index %d token %d val %d\n",
+			i, config->elems[i].token, config->elems[i].value);
+		switch(config->elems[i].token) {
+			case SOF_EXT_MAN_CONFIG_EMPTY:
+				/* unused memory space is zero filled - mapped to EMPTY elements */
+				break;
+			case SOF_EXT_MAN_CONFIG_IPC_MSG_SIZE:
+				/* TODO: use ipc msg size from config data */
+				break;
+			default:
+				dev_warn(sdev->dev, "Unknown firmware configuration token %d value %d",
```

Comment:
@ktrzcinx I'd like this to be dev_info() to align with https://github.com/thesofproject/linux/pull/2519

Reply:
changed to dbg message